ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,576 @@
+
+::
+
+  -- transactionally remove campaign 46 and all its ads
+
+  BEGIN;
+  DELETE from campaigns where id = 46 AND company_id = 5;
+  DELETE from ads where campaign_id = 46 AND company_id = 5;
+  COMMIT;
+
+As a final demonstration of SQL support, this query includes a joins, grouping, aggregation and a window function, and it executes fine. The query ranks the ads in each campaign by the count of their impressions.
+
+::
+
+  SELECT a.campaign_id,
+         RANK() OVER (
+           PARTITION BY a.campaign_id
+           ORDER BY a.campaign_id, count(*) desc
+         ), count(*) as n_impressions, a.id
+    FROM ads as a,
+         impressions as i
+   WHERE a.company_id = 5
+     AND i.company_id = a.company_id
+     AND i.ad_id      = a.id
+  GROUP BY a.campaign_id, a.id
+  ORDER BY a.campaign_id, n_impressions desc;
+
+In short when queries are scoped to a tenant then inserts, updates, deletes, complex SQL, and transactions all work as expected.
+
+Sharing Data Between Tenants
+----------------------------
+
+Up until now all tables have been distributed by :code:`company_id`, but sometimes there is data that can be shared by all tenants, and doesn't "belong" to any tenant in particular. For instance all companies using this example ad platform might want to get geographical information for their audience based on IP addresses. In a single machine database this could be accomplished by a lookup table for geo-ip, like the following. (A real table would probably use PostGIS but bear with the simplified example.)
+
+::
+
+  CREATE TABLE geo_ips (
+    addrs cidr NOT NULL PRIMARY KEY,
+    latlon point NOT NULL
+      CHECK (-90  <= latlon[0] AND latlon[0] <= 90 AND
+             -180 <= latlon[1] AND latlon[1] <= 180)
+  );
+  CREATE INDEX ON geo_ips USING gist (addrs inet_ops);
+
+Joining clicks with this table can produce, for example, the locations of everyone who clicked on ad 456.
+
+::
+
+  SELECT latlon
+    FROM geo_ips, clicks c
+   WHERE addrs >> c.user_ip
+     AND c.clicked_at > current_date - INTERVAL '1 day'
+     AND c.company_id = 5
+     AND c.ad_id = 456;
+
+To run this query efficiently in a distributed setup, we need to find a way to co-locate the :code:`geo_ips` table with clicks for every company. That way, no network traffic need be incurred at query time. One way would be to add a :code:`company_id` column to :code:`geo_ips` and duplicate the data in the table for every company. This approach is not optimal because it introduces the burden of keeping the data synchronized between the companies if and when it changes. A more convenient way is by designating :code:`geo_ips` as a *reference table.*
+
+Reference tables are replicated across all worker nodes, and Citus automatically keeps them in sync during modifications. This co-locates the information for all tenants' queries. To make a reference table, call :ref:`create_reference_table <create_reference_table>` for a table on the coordinator node:
```

Comment:
could we add an "instead of create_distributed_table" to the end so that users don't end up running this after running create_distributed_table.